lib/MooX/Options/Role.pm

```diff
@@ -278,13 +278,13 @@ sub parse_options {
     # create usage str
     my $usage_str = $options_config{usage_string} // "USAGE: $prog_name %o";
 
-    my ( $opt, $usage ) = describe_options(
-        ($usage_str), @$options,
-        [ 'usage', 'show a short help message'],
-        [ 'help|h', "show a help message" ],
-        [ 'man', "show the manual" ],
-        ,@flavour
-    );
+    unshift @$options, $options_config{usage_top}          if $options_config{usage_top};
```

Comment:
What is this ? is it an option ?
To push it on top, use "order" params. order => -1 = on top, order => 999 = on bottom.

If you want to add test before the usage, and after, that should be another kind of patch.